dbt/adapters/spark/relation.py

```diff
@@ -27,6 +32,8 @@ class SparkRelation(BaseRelation):
     quote_character: str = "`"
     is_delta: Optional[bool] = None
     is_hudi: Optional[bool] = None
+    is_iceberg: Optional[bool] = None
+    # TODO: make this a dict everywhere
```

Comment:
@mikealfare I know you called this out elsewhere. I tried making it a Dict. Problem is, we use some regex logic that wants this to be a nonquoted string and I don't think we should hold up this specific PR with what should be a more general PR on refactoring spark.